src/common/MessageProcessor.sol

```diff
@@ -68,11 +70,12 @@ contract MessageProcessor is Auth, IMessageProcessor {
     /// @inheritdoc IMessageHandler
     function handle(uint16 centrifugeId, bytes calldata message) external auth {
         MessageType kind = message.messageType();
-        uint16 sourceCentrifugeId = message.messageSourceCentrifugeId();
 
+        uint16 sourceCentrifugeId = message.messageSourceCentrifugeId();
         require(sourceCentrifugeId == 0 || sourceCentrifugeId == centrifugeId, InvalidSourceChain());
 
         if (kind == MessageType.ScheduleUpgrade) {
+            require(centrifugeId == MAINNET_CENTRIFUGE_ID, OnlyFromMainnet());
             MessageLib.ScheduleUpgrade memory m = message.deserializeScheduleUpgrade();
             root.scheduleRely(m.target.toAddress());
         } else if (kind == MessageType.CancelUpgrade) {
```

Comment:
We should also enforce only `MAINNET_CENTRIFUGE_ID` for `CancelUpgrade`, otherwise an exploited network can block all upgrades from mainnet.

Reply:
Actually, we should also only allow `MAINNET_CENTRIFUGE_ID` for `RecoverTokens`. Again, otherwise a rogue network can move all tokens.

Reply:
In the same direction: how do we prevent against a rogue network doing arbitrary `SetPoolAdaptersManager` calls for other pools? 🤔

Reply:
We're protected actually, thanks to `messageSourceCentrifugeId`, because it's checked that the pool used is received from the correct hub chain. So that's a reason not to remove that.

The same applies to `SetPoolAdapters` message.

Reply:
> The same applies to SetPoolAdapters message.

But this i snot true because of https://github.com/centrifuge/protocol-v3/blob/e776912409b25341deedff12cd74f9c215592621/src/common/libraries/MessageLib.sol#L169, right?

Reply:
(also, shouldn't we rename `SetPoolAdaptersManager` to `SetGatewayManager`?)